web/src/main.ts

```diff
@@ -1,21 +1,25 @@
+// Import external packages
 import Vue from 'vue';
-import VueCompositionAPI from '@vue/composition-api';
 import { sync } from 'vuex-router-sync';
 
 // @ts-ignore missing definitions
-import Girder, { vuetify } from '@girder/components/src';
+import { vuetify } from '@girder/components/src';
 import * as Sentry from '@sentry/browser';
 import * as Integrations from '@sentry/integrations';
 
-import App from '@/App.vue';
+// Import plugins first (order may matter)
+import '@/plugins/composition';
+import '@/plugins/girder';
+
+// Import custom behavior
 import '@/featureToggle';
+import '@/title';
+
+// Import internal items
+import App from '@/App.vue';
 import router from '@/router';
 import store from '@/store';
 import { girderRest, publishRest } from '@/rest';
-import '@/title';
-
-Vue.use(Girder);
-Vue.use(VueCompositionAPI);
```

Comment:
Actual question: what's the benefit of abstracting these into their own "plugin" files?

Reply:
This was somewhat necessary to preserve correct code-style, as well as allow for the feature toggles to use composition API. Since we need to call `Vue.use(VueCompositionAPI)` before using it anywhere, abstracting it into an import allows for those plugins to simply be imported, and then anything that uses them can come after. This is also generally applicable; Any features that may require plugins to be loaded first can just me imported/used after the plugin imports (which should probably be done first among local imports in `main.ts`).

Reply:
But the way it is now, the `Vue.use()` calls always happen exactly one time at application startup. Putting these calls into modules means that every import will cause a call to `Vue.use()` right? That may not be a problem, but I guess I don't understand how this leads to better code correctness.

For the record, I think even just the organization into logical units is a win. I just want to understand the other argument for doing it this way.

Reply:
In JS, imports only get evaluated once, so only the initial imports will cause `Vue.use()` to be called (in our case, on `main.ts` load). Regarding your other point, the reason this leads to code correctness is that otherwise, we'd have to import vue / composition-api, and then call `Vue.use`, then import`featureToggle` / whatever else would need to use it. Something like this:

```
import Vue from 'vue';
import VueCompositionAPI from '@vue/composition-api';

Vue.use(VueCompositionAPI);

import '@/featureToggle';
...
```

Which is generally bad code style, as imports should be at the top.

Reply:
Ah gotcha. Thanks for the explanation!

Reply:
For reference, a similar use in Viame-Web https://github.com/VIAME/VIAME-Web/tree/master/client/platform/web-girder/plugins